### PR TITLE
Document release procedure

### PR DESCRIPTION
**Pull request by mrocklin**
These are my current steps to release dask.  I thought it'd be good to add them as a PR for two reasons

1.  So that others can release
2.  In case we want to change this procedure.

When acting alone I avoid unnecessary process.  Dask may have matured beyond this though in some cases.  It would be nice to open up discussion on how we release.

### CODE REVIEW

File: docs/release-procedure.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    and conda.recipe/meta.yaml (if present).  Commit.
+
+*   Tag commit
+
+        git tag -a x.x.x -m 'Version x.x.x'
+
+*   and push to github
+
+        git push blaze master
+        git push blaze --tags
```
Comment (cpcloud) [inline docs/release-procedure.md]: You can do this in one step.

**Comment by stonebig** [comment]
````
python setup.py sdist bdist_wheel  --universal
twine upload dist/*
````

File: docs/release-procedure.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+*  Upload to PyPI
+
+        python setup.py register sdist upload
+
+*   Update anaconda recipe.
+
+    This should happen automatically within a day or two.
```
Comment (cowlicks) [inline docs/release-procedure.md]: Is this referring to changing the package in the conda repos?

Comment (mrocklin) [inline docs/release-procedure.md]: Yes

Comment (cowlicks) [inline docs/release-procedure.md]: I didn't know that was automatic, I thought we had to bug the conda team.

File: docs/release-procedure.md (hunk 1 above)
Comment (cowlicks) [inline docs/release-procedure.md]: `git push blaze master --tags` ?

It is weird that the tags are not "pushed" with the branch.

File: docs/release-procedure.md (hunk 2 above)
Comment (cowlicks) [inline docs/release-procedure.md]: @mrocklin We don't actually have to bug the conda team? It is automatic?

Comment (mrocklin) [inline docs/release-procedure.md]: Yes

**Comment by cowlicks** [comment]
What about something more general to start with like:

Reach consensus on what should be included in the release, and what can wait for the next release. This  usually happens in internal conversations. 

We also would benefit from some extra testing steps. Like:

Test builds on Windows.

**Pull request merged**